python/ray/tune/examples/pbt_tune_cifar10_with_keras.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+'''Train keras CNN on the CIFAR10 small images dataset.
+
+The model comes from: https://zhuanlan.zhihu.com/p/29214791,
+and it gets to about 87% validation accuracy in 100 epochs.
+
+Note that the scipt cannot init cuda in parallel, hence it
+only runs in cpu mode and might be time-consuming.
+'''
+
+from __future__ import print_function
+
+import argparse
+import os
+
+import numpy as np
+import tensorflow as tf
+from tensorflow.python.keras.backend import set_session
+from tensorflow.python.keras.datasets import cifar10
+from tensorflow.python.keras.layers import Input, Dense, Dropout, Flatten
+from tensorflow.python.keras.layers import Convolution2D, MaxPooling2D
+from tensorflow.python.keras.models import Model
+from tensorflow.python.keras.preprocessing.image import ImageDataGenerator
+
+import ray
+from ray.tune import grid_search, run_experiments
+from ray.tune import register_trainable
+from ray.tune import Trainable
+from ray.tune import TrainingResult
+from ray.tune.pbt import PopulationBasedTraining
+
+
+config = tf.ConfigProto(log_device_placement=True)
```

Comment:
Because tf.keras share a global setter, I'm not sure whether it is a good idea to place the config below main. I prefer to placing it in `_setup` method, however if ray don't allow assign one same GPU to different trials, I think the config is useless and can be removed.